src/datatype.rs

```diff
@@ -749,201 +752,140 @@ fn lookup_global_promote(lookup_name: &str) -> Option<(Args, Datatype)> {
     None
 }
 
-fn lookup_global_function(lookup_name: &str) -> Option<(Args, Datatype)> {
-    let global_functions = match Game::game() {
+fn global_function(lookup_name: &str) -> Option<(Args, Datatype)> {
+    let global_functions_map = match Game::game() {
         #[cfg(feature = "ck3")]
-        Game::Ck3 => crate::ck3::tables::datafunctions::GLOBAL_FUNCTIONS,
+        Game::Ck3 => &crate::ck3::tables::datafunctions::GLOBAL_FUNCTIONS_MAP,
         #[cfg(feature = "vic3")]
-        Game::Vic3 => crate::vic3::tables::datafunctions::GLOBAL_FUNCTIONS,
+        Game::Vic3 => &crate::vic3::tables::datafunctions::GLOBAL_FUNCTIONS_MAP,
         #[cfg(feature = "imperator")]
-        Game::Imperator => crate::imperator::tables::datafunctions::GLOBAL_FUNCTIONS,
+        Game::Imperator => &crate::imperator::tables::datafunctions::GLOBAL_FUNCTIONS_MAP,
     };
-    if let Ok(idx) = global_functions.binary_search_by_key(&lookup_name, |(name, _, _)| name) {
-        let (_name, args, rtype) = global_functions[idx];
-        return Some((args, rtype));
-    }
-    None
+    global_functions_map.get(lookup_name).copied()
 }
 
-fn lookup_promote_or_function(
-    lookup_name: &str,
-    ltype: Datatype,
-    global: &[(&str, Datatype, Args, Datatype)],
-) -> LookupResult {
-    let start = global.partition_point(|(name, _, _, _)| name < &lookup_name);
-    let mut found_any = false;
+#[inline]
+fn lookup_global_function(lookup_name: &str) -> Option<(Args, Datatype)> {
+    global_function(lookup_name)
+}
```

Comment:
Why these tiny functions? I don't understand why `lookup_global_function` and `global_function` are separate

Reply:
This is purely to mirror promotes and functions, but yeah i think they should be removed